File: Vanilla/sources/zomboid/zombie/ai/states/IdleState.java
package zombie.ai.states;

import zombie.ai.State;
import zombie.characters.IsoGameCharacter;
import zombie.core.skinnedmodel.advancedanimation.AnimEvent;
import zombie.util.StringUtils;

public final class IdleState extends State {
   private static final IdleState _instance = new IdleState();

   public static IdleState instance() {
      return _instance;
   }

   public void animEvent(IsoGameCharacter var1, AnimEvent var2) {
      if (var2.m_EventName.equalsIgnoreCase("PlaySound") && !StringUtils.isNullOrEmpty(var2.m_ParameterValue)) {
         var1.getSquare().playSound(var2.m_ParameterValue);
      }

   }
}

File: Vanilla/sources/zdoc/media/lua/shared/Library/IdleState.lua
---@class IdleState : zombie.ai.states.IdleState
---@field private _instance IdleState
IdleState = {}

---@public
---@param arg0 IsoGameCharacter
---@param arg1 AnimEvent
---@return void
function IdleState:animEvent(arg0, arg1) end

---@public
---@return IdleState
function IdleState:instance() end
